CppCoreGuidelines.md

```diff
@@ -2847,9 +2849,9 @@ When copying is cheap, nothing beats the simplicity and safety of copying, and f
 
 For advanced uses (only), where you really need to optimize for rvalues passed to "input-only" parameters:
 
-* If the function is going to unconditionally move from the argument, take it by `&&`. See [F.18](#Rf-consume).
+* If the function is going to unconditionally move from the argument, take it by `&&`. See [F.18](#Rf-consume). You may also provide a `const &` overload that copies the argument rather than moves it, although this is not required.
```

Comment:
Since this is under the text *For [...] where you really need to optimize for rvalues [...]* I think this change is redundant.

Reply:
This came from a specific point I was confused by, which is that the guidelines allow you to provide only an `&&` overload in these situations.  To me, this is a useful clarification since it explicitly says that you're not required to provide a `const &` overload as well.

Reply:
Based on the ```shared_ptr``` guidance in R.34 (and the discussion in #1989), should F.18 be updated to allow pass by value and then move?

    // Follows F.18
    B convert(A&& a) {
      return B{std::move(a).data};
    }
or

    // Also follows F.18
    B convert(A a) {
      return B{std::move(a).data};
    }

Pass by value is simpler than ```const&``` and ```&&``` overloads but does result in an extra move.  I sometimes pass vectors this way in my own code.  F.18 already has an exception for ```unique_ptr```, should another exception be added for cheap to move types?

Reply:
Those are two different cases: A&& forces a r- value to be passed (this is the classic "consume" semantic. A is a simplified version of the `const A&` and `A&&` overload set, which means I will keep a copy.

Reply:
I cant move from a const&, so this change seems wrong to me.

Reply:
What this comment meant to convey is - If you wrote a move-only sink function, it must use `&&`.  However, it may be more convenient for the caller to provide both a `&&` and `const &` overload, and this is indeed allowed by the guidelines.  As written I've seen many people be confused by this note, since they thought since they wrote a function that always moves an argument, they are somehow not allowed to provide a `const &` overload.  They probably got this impression because they only read this note and did not perform a close reading of the full set of guidelines `F.16-F.18`. That's the confusion I'm trying to avoid by editing this note.  Could you suggest a better way to make this clarification?

Reply:
Regarding @bgloyer 's suggestion to change the guidelines to allow pass-by-value, I'm in support of this but this would be more disruptive than the change I was trying to make on this PR.  In particular, I'm a big fan of the alternative function passing guidelines proposed [here](https://github.com/isocpp/CppCoreGuidelines/issues/526#issuecomment-779820798).

Reply:
I guess the first question is should the F.call guidelines be updated to allow pass by value and move if the callee consumes the parameter?  Then if so, where should it go?